### ISSUE TITLE
bug: Nushell error messages remains even it is not enabled, and home-manager-user.service fails

### ISSUE THREAD

**Issue by MarkGhanz** [nix-community/home-manager#5179]
### Are you following the right branch?

- [X] My Nixpkgs and Home Manager versions are in sync

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue description

When restarting home-manager-user.service after `nixos-rebuild switch`, 
```
3月 25 10:12:37 MKZFRMW hm-activate-karm2[8975]: Error:   × Nushell launched as a REPL, but STDIN is not a TTY; either launch in a
3月 25 10:12:37 MKZFRMW hm-activate-karm2[8975]:   │ valid terminal or provide arguments to invoke a script!
```
still appears even if `programs.nushell.enabled` is unset.

### System information

```markdown
- system: `"x86_64-linux"`
 - host os: `Linux 6.8.1, NixOS, 24.05 (Uakari), 24.05.20240323.44d0940`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `"nixos, nixos-hardware"`
 - nixpkgs: `/nix/store/amxd2p02wx78nyaa4bkb0hjvgwhz1dq7-source`
```

**Comment by rachitvrma** [comment]
Facing the same issue.

**Comment by wthueb** [comment]
Just ran into this with my setup and figured it out for me. I use nix-darwin and NixOS, and MacOS doesn't play nicely with having nushell as the user's login shell. To get around this, I do `programs.bash.profileExtra = "exec nu";` with `bash` as the default shell. However, `home-manager-*.service` uses a login bash shell under your user, so it'll run the profile and thus exec `nu`, causing this error.

TLDR make sure you don't exec nu in your profile, whether that be in `/etc/profile` or `~/.profile`.
